packages/ui/.storybook/components/SelectedColorPalette.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { ColorPalette, ColorItem } from '@storybook/addon-docs/blocks';
+import { colors } from '../../src/theme/overwrites/foundations/colors';
+interface SelectedColorPaletteProps {
+  color: string;
+}
+
+export const SelectedColorPalette = ({color}: SelectedColorPaletteProps) => {
+  const allColorVariantsInOneItem = (color: string) => {
+    return <ColorItem title='' subtitle='' colors={Object.values(colors[color])} />;
+  };
+
+  const eachColorVariantInSeparateItem = (color: string) => {
+    const selectedColor = colors[color];
+    return Object.keys(selectedColor).map((value) => (
+      <ColorItem title={`${color}.${value}`} subtitle={selectedColor[value]} colors={[selectedColor[value]]} />
+    ));
+  };
```

Comment:
💭 Those functions don't need to be inside this component so you can move them outside of it. Also as they return React elements it could be a great idea to create components from them - e.g. AllVariants, SingleVariants (you could keep them in the same file).